pkg/download/buffer.go

```diff
@@ -37,12 +37,30 @@ func (t *Target) Basename() string {
 	return filepath.Base(t.Dest)
 }
 
-func getBufferMode(client *client.HTTPClient) Mode {
+func getBufferMode(opts Options) Mode {
+	client := client.NewHTTPClient(opts.Client)
 	return &BufferMode{
-		Client: client,
+		Client:  client,
+		Options: opts,
 	}
 }
 
+func (m *BufferMode) maxChunks() int {
+	maxChunks := m.MaxChunks
+	if maxChunks == 0 {
+		return runtime.GOMAXPROCS(0) * 4
```

Comment:
Can we move this to NumCPU() instead?

```suggestion
		return runtime.NumCPU(0) * 4
```

Or should we use the automaxprocs?